Antidote/CallsManager.m

```diff
@@ -451,13 +465,33 @@ - (void)switchViewControllerForCall:(OCTCall *)call
 
     self.currentCallViewController = abstractVC;
 
-    self.currentCall = call;
+    self.currentCall = [RBQSafeRealmObject safeObjectFromObject:call];
 }
 
-- (BOOL)onlyPauseCallsLeft
+- (void)replaceCurrentCallThatWasRemoved
 {
-    return (([self.allActiveCallsController.fetchedObjects count] == 0) &&
-            ([self.allPausedCallsController.fetchedObjects count] != 0));
+    OCTCall *call = [self.allPausedCallsController.fetchedObjects firstObject];
+
+    if (! call) {
+        call = [self.allCallsController.fetchedObjects firstObject];
+    }
+
+    // workaround for deadlock in objcTox
+    // https://github.com/Antidote-for-Tox/objcTox/issues/51
+    [self performSelector:@selector(switchViewControllerForCall:) withObject:call afterDelay:0];
```

Comment:
Yeah, I understand that. What I meant was to check once again if there is no call at all before calling `performSelector...`. In this case CallsManager should be dead, but it would nice to have safer code.

```
OCTCall *call = [self.allPausedCallsController.fetchedObjects firstObject] ?:
                [self.allCallsController.fetchedObjects firstObject];

if (! call) {
    return;
}

[self performSelector:...];
```